consensus/state.go

```diff
@@ -1922,6 +1950,7 @@ func (cs *State) addProposalBlockPart(msg *BlockPartMessage, peerID p2p.ID) (add
 
 	added, err = cs.ProposalBlockParts.AddPart(part)
 	if err != nil {
+		// TODO: can we punish for these or is there a reason they might accidentaly happen?
 		if errors.Is(err, types.ErrPartSetInvalidProof) || errors.Is(err, types.ErrPartSetUnexpectedIndex) {
 			cs.metrics.BlockGossipPartsReceived.With("matches_current", "false").Add(1)
```

Comment:
Can this happen during catchup?

Reply:
> Can this happen during catchup?

The catch-up consists of receiving block parts from a peer. As mentioned above (3), if we don't know which block the peer is looking for (cases 1 or 2), we ship to the peer the block we have committed on that height.